sunpy/map/tests/test_mapbase.py

```diff
@@ -686,8 +686,15 @@ def test_resample_metadata(generic_map, sample_method, new_dimensions):
     Check that the resampled map has correctly adjusted metadata.
     """
     resampled_map = generic_map.resample(new_dimensions, method=sample_method)
-    assert resampled_map.meta['cdelt1'] == generic_map.meta['cdelt1']
-    assert resampled_map.meta['cdelt2'] == generic_map.meta['cdelt2']
+
+    scale_x = generic_map.data.shape[1] / resampled_map.data.shape[1]
+    scale_y = generic_map.data.shape[0] / resampled_map.data.shape[0]
+
+    assert resampled_map.meta['cdelt1'] == scale_x * generic_map.meta['cdelt1']
+    assert resampled_map.meta['cdelt2'] == scale_y * generic_map.meta['cdelt2']
+    assert resampled_map.meta['pc1_2'] == scale_y / scale_x * generic_map.meta['pc1_2']
+    assert resampled_map.meta['pc2_1'] == scale_x / scale_y * generic_map.meta['pc2_1']
```

Comment:
```suggestion
    assert resampled_map.meta['pc2_1'] == scale_x / scale_y * generic_map.meta['pc2_1']
    assert resampled_map.meta['pc1_1'] == generic_map.meta['pc1_1']
    assert resampled_map.meta['pc2_2'] == generic_map.meta['pc2_2']
```
Worth adding a check that the pc1_1 and pc2_2 metadata hasn't been changed? Even though it's implicitly checked below, I think for clarity it's worth having an explicit check up here.

Reply:
Sure, done